lib/engine/game/g_1824/game.rb

```diff
@@ -718,8 +898,128 @@ def float_corporation(corporation)
+            city.hex.tile.reservations.delete(ps)
+            home_token.remove!
+            if state_railway.tokens.find { |t| t.city == city }
+              @log << "#{ps.name} token in #{city.hex.name} is removed as #{state_railway.name} already have a token there"
+            else
+              @log << "#{ps.name} token in #{city.hex.name} is replaced with a #{state_railway.name} token"
+              city.place_token(state_railway, state_railway.next_token, free: true, check_tokenable: false)
+            end
+
+            @log << "#{ps.name} is now closed"
+            ps.close!
+          end
+
+          # Add player in priority order to presidency possible tie breaker
+          presidency_priority_order |= @players
+
+          floating_capital = (120 * 10) - pre_states_value
+          @bank.spend(floating_capital, state_railway)
+          capital_str = format_currency(floating_capital)
+          @log << "#{state_railway.name} receives floating capital #{capital_str} for all non-reserved shares"
+          # Check presidency according to rules
+          max_shares = @share_pool.presidency_check_shares(state_railway).values.max
+          if max_shares < 20
+            # Receivership - will not act, just fall in stock price, until someone takes the role
+            @log << "#{state_railway.name} has no president. It will decrease in value each OR until someone purchases up to 20%"
+            state_railway.owner = @share_pool
+            presidency_share = state_railway.shares.find(&:president)
+            presidency_share.buyable = false
+            # The president share will be given to the first player that has 2 10% shares
+          else
+            majority_share_holders = @share_pool.presidency_check_shares(state_railway).select { |_, p| p == max_shares }.keys
+            actual_president = majority_share_holders.max_by { |s| presidency_priority_order.find_index(s) }
+            @log << "#{actual_president.name} becomes president for #{state_railway.name}"
+            presidency_share = state_railway.presidents_share
+            if presidency_share.owner != actual_president
+              @share_pool.change_president(presidency_share, presidency_share.owner, actual_president)
+            end
+            state_railway.owner = actual_president
+          end
+          @log << "#{state_railway.name} floats and will operate next OR"
+          state_railway.floatable = true
+          state_railway.floated = true
+        end
+
+        def post_train_buy(train)
+          case train.name
+          when '2'
+            unless @two_train_bought
+              @two_train_bought = true
+              @log << '-- Event: 1g trains now available'
+            end
+          when '3g', '4g', '5g'
+            return if @rustee_gtrains.include?(train.name)
+
+            @rustee_gtrains << train.name
+            rust_trains!(train, @current_entity)
+          end
+        end
+
```

Comment:
the code has events that can be triggered when a train is bought or on certain phases.
so if you define event "1g_available", you can in def of def event_1g_available! 
set the boolean and print to logs